dimagi/utils/couch/migration.py

```diff
@@ -135,10 +136,10 @@ def _migration_get_couch_model_class(cls):
         raise NotImplementedError()
 
     def _migration_get_couch_object(self):
-        if not self.couch_id:
+        if not getattr(self, self.couch_id_name, None):
```

Comment:
also i don't think we should default this to None, we should fail hard if couch_id_name is not a property